### PR TITLE
Update to Micronaut 4 and Gradle 8

### LINKED ISSUE #4: Upgrade to Mn4 and Gradle 4

### PR DESCRIPTION
**Pull request by timyates**
This PR moves to use Micronaut 4.1.0 and Gradle 8.3

It removes the serverless runtime for testing, and instead uses our AWS testing

There are some outstanding issues around https://github.com/micronaut-projects/micronaut-lambda-todo/commit/eed9e34ce9f377a42af492a98a91e8c94ad9b9b1

And the order of attributes in introspection has changed, so I needed to switch to https://github.com/micronaut-projects/micronaut-lambda-todo/commit/b6b60513d71f2e5d9d4544959adb5f5cee7caed8 when rebuilding objects loaded from the database

### CODE REVIEW

**Comment by timyates** [comment]
Failing on CI and passing locally... investigating

File: code/src/main/java/com/micronauttodo/apigateway/DefaultStageResolver.java [hunk 1]
```diff
@@ -11,11 +9,11 @@
...
+
     @Override
     @NonNull
     public Optional<String> resolveStage(@Nullable HttpRequest<?> request) {
-        return Optional.ofNullable(request)
-                .flatMap(req -> req.getAttribute(RequestReader.API_GATEWAY_CONTEXT_PROPERTY, AwsProxyRequestContext.class))
-                .map(AwsProxyRequestContext::getStage);
+        // TODO: What is this??
+        return Optional.empty();
     }
```
Comment (timyates) [inline code/src/main/java/com/micronauttodo/apigateway/DefaultStageResolver.java]: I'm not sure what this used to do...

File: infra/src/test/java/com/example/AppStackTest.java [hunk 2]
```diff
@@ -17,7 +17,7 @@ void testAppStack() {
                 new File(AppStack.functionPath("function-java", Runtime.JAVA)).exists()) {
             AppStack stack = new AppStack(new App(), "TestMnTodoStack");
             Template template = Template.fromStack(stack);
-            template.hasResourceProperties("AWS::Lambda::Function", Collections.singletonMap("Handler", "io.micronaut.function.aws.proxy.MicronautLambdaHandler"));
+            template.hasResourceProperties("AWS::Lambda::Function", Collections.singletonMap("Handler", "io.micronaut.function.aws.proxy.payload1.ApiGatewayProxyRequestEventFunction"));
```
Comment (timyates) [inline infra/src/test/java/com/example/AppStackTest.java:20]: @sdelamo Are we missing an assertion here, as this wasn't failing the build?

Comment (sdelamo) [reply infra/src/test/java/com/example/AppStackTest.java:20]: ./gradlew infra:test was failing without this change

Comment (timyates) [reply infra/src/test/java/com/example/AppStackTest.java:20]: Weird... was that not run by the action?

**Pull request merged**